internal/p2p/peermanager.go

```diff
@@ -1726,3 +1736,21 @@ func SetProTxHashToPeerInfo(proTxHash types.ProTxHash) func(info *peerInfo) {
 		info.ProTxHash = proTxHash.Copy()
 	}
 }
+
+// evictPeerAfterTimeout evicts incoming peer for which the timeout expired.
+func evictPeerAfterTimeout(m *PeerManager, peerID types.NodeID, direction peerConnectionDirection, timeout time.Duration) {
+	if timeout > 0 {
+		time.AfterFunc(timeout, func() {
+			olderThan := time.Now().Add(-timeout)
+
+			m.mtx.Lock()
```

Comment:
it would be better if you create a method `GetPeer` at `PeerManager` and move this logic there